x-pack/plugin/identity-provider/docs/en/rest-api/idp-saml-init.asciidoc

```diff
@@ -0,0 +1,122 @@
+[role="xpack"]
+[[idp-saml-init]]
+=== Generate a SAML authentication response message
+++++
+<titleabbrev>Generate SAML response</titleabbrev>
+++++
+Generates a SAML Response message to be sent to a Service Provider
+
+[[idp-saml-init-request]]
+==== {api-request-title}
+
+`POST /_idp/saml/init`
+
+[[idp-saml-init-prereqs]]
+==== {api-prereq-title}
+
+* To use this API, you must have a role that grants the `cluster:admin/idp/saml/init` privilege.
+
+[[idp-saml-init-desc]]
+==== {api-description-title}
+
+This API generates a SAML Response message that should be sent to a Service Provider as part of an
+IDP initiated or SP initiated SAML Single Sign On. This API expects the caller to present
+credentials for the user that the SAML Response will be created for as "Secondary Authentication"
+using the `es-secondary-authorization` HTTP Request header.
```

Comment:
FYI, @lcawl is working on some generic docs for secondary authentication, so which we can link to when they're ready.